analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -0,0 +1,163 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp;
+
+public sealed class SecureRandomSeedsShouldNotBePredictable : SymbolicRuleCheck
+{
+    private const string DiagnosticId = "S4347";
+    private const string MessageFormat = "Set an unpredictable seed before generating random values.";
+    private static readonly ImmutableHashSet<string> SecureRandomNextMethods = ImmutableHashSet.Create(
+               "Next",
+               "NextInt",
+               "NextLong",
+               "NextDouble",
+               "NextBytes");
+
+    public static readonly DiagnosticDescriptor S4347 = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+    protected override DiagnosticDescriptor Rule => S4347;
+
+    public override bool ShouldExecute() => true;
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        var operation = context.Operation.Instance;
+
+        if (operation.AsArrayCreation() is { } arrayCreation)
+        {
+            return ProcessArrayCreation(state, arrayCreation);
+        }
+        else if (operation.AsArrayElementReference() is { } arrayElementReference)
+        {
+            return ProcessArrayElementReference(state, arrayElementReference);
+        }
+        else if (operation.AsInvocation() is { } invocation)
+        {
+            return ProcessArraySetValue(state, invocation)
+                ?? ProcessStringToBytes(state, invocation)
+                ?? ProcessSecureRandomGetInstance(state, invocation)
+                ?? ProcessSecureRandomSetSeed(state, invocation)
+                ?? ProcessSecureRandomNext(state, invocation)
+                ?? state;
+        }
+        return state;
+    }
+
+    private static ProgramState ProcessArrayCreation(ProgramState state, IArrayCreationOperationWrapper arrayCreation)
+    {
+        if (arrayCreation.Type.IsAny(KnownType.System_Byte_Array, KnownType.System_Char_Array))
+        {
+            var isConstant = arrayCreation.Initializer.WrappedOperation is null || arrayCreation.Initializer.ElementValues.All(x => x.ConstantValue.HasValue);
+            return state.SetOperationConstraint(
+                arrayCreation,
+                isConstant ? CryptographicSeedConstraint.Predictable : CryptographicSeedConstraint.Unpredictable);
+        }
+        return state;
+    }
+
+    private static ProgramState ProcessArrayElementReference(ProgramState state, IArrayElementReferenceOperationWrapper arrayElementReference) =>
+        arrayElementReference.IsAssignmentTarget() || arrayElementReference.IsCompoundAssignmentTarget()
+            ? state.SetSymbolConstraint(arrayElementReference.ArrayReference.TrackedSymbol(state), CryptographicSeedConstraint.Unpredictable)
```

Comment:
Here, if the array access is on the left side of the assignment, I set it to unpredictable.
I could look into the right side of the assignment and see if it has constant value, but I am not sure if there is much value in this case, so I opted for the FN.